ledger/acctupdates.go

```diff
@@ -1918,38 +1903,36 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 	au.accountsMu.Lock()
 	// Drop reference counts to modified accounts, and evict them
 	// from in-memory cache when no references remain.
-	for addr, cnt := range flushcount {
+	for addr, acctDltCnt := range compactDeltas {
+		cnt := acctDltCnt.ndeltas
 		macct, ok := au.accounts[addr]
 		if !ok {
 			au.log.Panicf("inconsistency: flushed %d changes to %s, but not in au.accounts", cnt, addr)
 		}
 
 		if cnt > macct.ndeltas {
 			au.log.Panicf("inconsistency: flushed %d changes to %s, but au.accounts had %d", cnt, addr, macct.ndeltas)
-		}
-
-		macct.ndeltas -= cnt
-		if macct.ndeltas == 0 {
+		} else if cnt == macct.ndeltas {
 			delete(au.accounts, addr)
 		} else {
+			macct.ndeltas -= cnt
 			au.accounts[addr] = macct
 		}
 	}
 
-	for cidx, cnt := range creatableFlushcount {
+	for cidx, modCrt := range compactCreatableDeltas {
+		cnt := modCrt.ndeltas
 		mcreat, ok := au.creatables[cidx]
 		if !ok {
 			au.log.Panicf("inconsistency: flushed %d changes to creatable %d, but not in au.creatables", cnt, cidx)
 		}
 
 		if cnt > mcreat.ndeltas {
 			au.log.Panicf("inconsistency: flushed %d changes to creatable %d, but au.creatables had %d", cnt, cidx, mcreat.ndeltas)
-		}
-
-		mcreat.ndeltas -= cnt
-		if mcreat.ndeltas == 0 {
+		} else if cnt == mcreat.ndeltas {
 			delete(au.creatables, cidx)
 		} else {
+			mcreat.ndeltas -= cnt
 			au.creatables[cidx] = mcreat
```

Comment:
Same as above, what is the purpose of this assignment? Isn't mcreat already in au.creatables[cidx]?

Reply:
Same as above